contracts/mainchain/MainchainGatewayV2.sol

```diff
@@ -298,18 +303,24 @@ contract MainchainGatewayV2 is WithdrawalLimitation, Initializable, AccessContro
     address _tokenAddr = _receipt.mainchain.tokenAddr;
 
     _receipt.info.validate();
-    require(_receipt.kind == Transfer.Kind.Withdrawal, "MainchainGatewayV2: invalid receipt kind");
-    require(_receipt.mainchain.chainId == block.chainid, "MainchainGatewayV2: invalid chain id");
+    if (_receipt.kind != Transfer.Kind.Withdrawal) {
+      revert ErrInvalidReceiptKind();
+    }
+    if (_receipt.mainchain.chainId != block.chainid) {
+      revert ErrInvalidChainId(msg.sig);
+    }
+
```

Comment:
Change to in-line revert (remove curly braces when there is only revert)

Reply:
changed.